### PR TITLE
Rename ib-kubernetes deployment yaml

### PR DESCRIPTION
**Pull request by adrianchiris**
As we are not using a daemonset the suffix "ds" is missleading

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,5 +86,5 @@ To deploy the InfiniBand Kbubernetes
 ```shell script
 $ kubectl create -f deployment/ib-kubernetes-configmap.yaml
 $ kubectl create -f deployment/ib-kubernetes-ufm-secret.yaml
-$ kubectl create -f deployment/ib-kubernetes-ds.yaml
+$ kubectl create -f deployment/ib-kubernetes-deploy.yaml
```
Comment (moshe010) [inline README.md]: I think ib-kubernetes.yaml should be good enough

File: deployment/ib-kubernetes-deploy.yaml [hunk 2]
```diff
@@ -38,7 +38,7 @@ metadata:
   namespace: kube-system
   annotations:
     kubernetes.io/description: |
-      This daemonset launches the ib-kubernetes daemon for infiniband cni.
+      This deployment launches the ib-kubernetes daemon for infiniband CNI.
```
Comment (moshe010) [inline deployment/ib-kubernetes-deploy.yaml]: a/infiniband/InfiniBand

File: deployment/ib-kubernetes-deploy.yaml [hunk 3]
```diff
@@ -51,6 +51,10 @@ spec:
     spec:
       serviceAccountName: ib-kubernetes
       hostNetwork: true
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          operator: Exists
+          effect: NoSchedule
```
Comment (moshe010) [inline deployment/ib-kubernetes-deploy.yaml]: I think we need evaluate what is done in ovn-master yaml and see if it make sense for us [1]
[1] - https://github.com/ovn-org/ovn-kubernetes/blob/master/dist/templates/ovnkube-master.yaml.j2#L41-L65

File: deployment/ib-kubernetes-deploy.yaml (hunk 2 above)
Comment (moshe010) [reply deployment/ib-kubernetes-deploy.yaml]: we also should mark it as critical [1]
[1] -https://github.com/ovn-org/ovn-kubernetes/blob/master/dist/templates/ovnkube-master.yaml.j2#L33-L34

**Comment by adrianchiris** [comment]
marking a pod as critical is now done by `priorityClassName` pod attribute and not annotation (since 1.16)

**Pull request merged**